### PR TITLE
Feature/macos gfortran updates

### PR DESCRIPTION
**Pull request by rickgrubin**
Updates for building jedi-stack on macOS:

**Open MPI**

- add flags to manage hwloc and libevent
- see: https://www.open-mpi.org/faq/?category=building#libevent-or-hwloc-errors-when-linking-fortran

**GNU Fortran 10.2.0**

- add flag `-fallow-argument-mismatch` for pnetcdf

### CODE REVIEW

**Comment by rickgrubin** [comment]
**oops-bundle**
100% tests passed, 0 tests failed out of 439

Label Time Summary:
atlas         =  51.93 sec*proc (132 tests)
eckit         =  64.99 sec*proc (115 tests)
executable    = 135.55 sec*proc (284 tests)
fckit         =  17.39 sec*proc (14 tests)
fortran       =  11.91 sec*proc (34 tests)
mpi           =  30.50 sec*proc (36 tests)
oops          =  91.20 sec*proc (178 tests)
openmp        =  32.61 sec*proc (42 tests)
script        =  76.35 sec*proc (153 tests)

**ioda-engines**
100% tests passed, 0 tests failed out of 83

Label Time Summary:
ioda-engines    =   9.58 sec*proc (83 tests)

**ufo-bundle**
100% tests passed, 0 tests failed out of 675

Subproject Time Summary:
fckit    =  24.87 sec*proc (14 tests)
gsw      =   1.20 sec*proc (2 tests)
ioda     =  21.14 sec*proc (16 tests)
oops     = 110.39 sec*proc (180 tests)
saber    = 193.34 sec*proc (149 tests)
ufo      = 354.48 sec*proc (181 tests)

Label Time Summary:
atlas            =  90.68 sec*proc (133 tests)
download_data    =  16.20 sec*proc (2 tests)
executable       = 294.78 sec*proc (257 tests)
fortran          =  21.74 sec*proc (35 tests)
mpi              = 229.35 sec*proc (111 tests)
openmp           = 191.31 sec*proc (126 tests)
script           = 469.03 sec*proc (414 tests)

File: buildscripts/libs/build_pnetcdf.sh [hunk 1]
```diff
@@ -36,6 +36,10 @@ export CXX=$MPI_CXX
 export F9X=$FC
 
 export FFLAGS+=" -fPIC -w"
+# for GNU Fortran 10; see: https://github.com/Unidata/netcdf-fortran/issues/212#issuecomment-638457375
+[[ -n $FC ]] && FC_GFORTRAN_10=$($FC --version | grep -cE "GNU Fortran.* 1[0-9]\.[0-9]+")
```
Comment (mmiesch) [inline buildscripts/libs/build_pnetcdf.sh:40]: I'm running gfortran 9.3.0.  This line returns with an exit code of 1, which causes the script to exit:

```bash
11:30 $ [[ -n $FC ]] && FC_GFORTRAN_10=$($FC --version | grep -cE "GNU Fortran.* 1[0-9]\.[0-9]+")
(base) ✔ ~/JEDI/code/local_copy/jedi-stack/buildscripts [feature/macos_gfortran_updates|✚ 3…1] 
11:31 $ echo $?
1
(base) ✔ ~/JEDI/code/local_copy/jedi-stack/buildscripts [feature/macos_gfortran_updates|✚ 3…1] 
11:31 $ echo $FC
gfortran
```

Comment (mmiesch) [reply buildscripts/libs/build_pnetcdf.sh:40]: Oh - actually - it's exiting in the same line of the build_mpi.sh script - I haven't made it to pnetcdf yet

Comment (mmiesch) [reply buildscripts/libs/build_pnetcdf.sh:40]: It works if you just disable the exit
```
set +e
[[ -n $FC ]] && FC_GFORTRAN_10=$($FC --version | grep -cE "GNU Fortran.* 1[0-9]\.[0-9]+")
set -e
```

**Pull request merged**